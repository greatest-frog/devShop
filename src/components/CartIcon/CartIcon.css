img {
    filter: invert(7%) sepia(12%) saturate(353%) hue-rotate(22deg) brightness(95%) contrast(87%);
    -webkit-filter: invert(7%) sepia(12%) saturate(353%) hue-rotate(22deg) brightness(95%) contrast(87%);
}

.cart-icon {
    position: relative;
    bottom: -4px;
}

.goods-number {
    color: #f5f5f5;
    background: #1f1f1d;
    font-size: 14px;
    line-height: 18px;
    width: 18px;
    position: absolute;
    top: -7px;
    right: -5px;
    border-radius: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
}