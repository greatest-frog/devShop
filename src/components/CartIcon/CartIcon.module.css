.cartIcon {
    position: relative;
    bottom: -4px;
}

.cartIcon_wrapper {
    width: 24px;
    height: 24px;
    fill: var(--main-color);
}

.cartIcon__number {
    color: var(--background-color);
    background: var(--main-color);
    font-size: 14px;
    line-height: 18px;
    width: 18px;
    position: absolute;
    top: -7px;
    right: -5px;
    border-radius: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
}
