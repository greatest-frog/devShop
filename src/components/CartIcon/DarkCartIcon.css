.dark .cart-icon svg {
    fill: #f5f5f5;
}

.dark .goods-number {
    color: #1f1f1d;
    background: #f5f5f5;
}
