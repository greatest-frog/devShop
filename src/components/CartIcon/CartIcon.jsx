import "./CartIcon.css";

const CartIcon = ({ goodsNumber }) => {
  return (
    <div className="cart-icon" data-testid="cart-icon">
      <img src="./shop/resources/images/cart.png" alt="Cart" />
      <div className="goods-number">{goodsNumber}</div>
    </div>
  );
};

export default CartIcon;
