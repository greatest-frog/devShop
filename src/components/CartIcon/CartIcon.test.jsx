import CartIcon from "./CartIcon";
import { render, screen } from "@testing-library/react";

describe("Cart Icon", () => {
  test("Cart Icon renders number of items", () => {
    render(<CartIcon />);
    expect(screen.getByTestId("cart-icon")).toMatchSnapshot();
  });
});
