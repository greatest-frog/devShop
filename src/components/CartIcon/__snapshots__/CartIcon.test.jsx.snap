// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Icon Cart Icon renders number of items 1`] = `
<div
  class="cart-icon"
  data-testid="cart-icon"
>
  <img
    alt="Cart"
    src="./shop/resources/images/cart.png"
  />
  <div
    class="goods-number"
  />
</div>
`;
